src/models/MatrixAction.ts

```diff
@@ -179,7 +179,14 @@ export class MatrixAction {
                     fileSize = " (" + Math.round(event.content.info.size / 1024) + "KB)";
                 }
 
-                const url = ContentRepo.getHttpUriForMxc(mediaUrl, event.content.url);
+                let url = ContentRepo.getHttpUriForMxc(mediaUrl, event.content.url);
+                if (forceFilename) {
+                    url += `/${forceFilename}`;
+                }
+                else if (event.content.body?.match(/\.[\w\d]{2,4}$/)) {
```

Comment:
That's a pretty generous RegExp for filenames,

`I am not a file name / but I match 3.1415`
🙃

How about /^\S\.[\w\d]{2,4}$` to rule out whitespaces (yes, some of which are legal in filenames)?
Well, honestly... whatever. This is trying to simplify very complicated, OS-dependent rules. 🙈

Reply:
Yep, I understand. This is just a very simple heuristic to get a filename for the file if we can. I am not terribly happy that the spec doesn't have an explicit field for filename and caption, as some clients use `body` for descriptive text.

Reply:
On the other hand, having a filename with spaces in it will *only* mean we don't bother with a file name, which isn't that common anyway. Let's use yours.